bbs/input.h

```diff
@@ -54,7 +54,7 @@ std::string input_password(const std::string& prompt_text, int max_length);
 template<typename T>
 typename std::enable_if<std::is_signed<T>::value, T>::type
 input_number(T current_value, T min_value, T max_value, bool set_default_value = true) {
-  int len = std::max<int>(1, static_cast<T>(ceil(log10(max_value))));
+  int len = std::max<int>(1, static_cast<T>(ceil(log10((double)max_value))));
```

Comment:
Using GCC 4.9.4

It wasn't compiling as it was complaining that it was ambigious:

```
In file included from /home/andrew/wwiv/bbs/conf.cpp:27:0:
/home/andrew/wwiv/bbs/../bbs/input.h: In instantiation of ‘typename std::enable_if<std::is_unsigned<_Tp>::value, T>::type input_number(T, T, T, bool) [with T = unsigned char; typename std::enable_if<std::is_unsigned<_Tp>::value, T>::type = unsigned char]’:
/home/andrew/wwiv/bbs/conf.cpp:559:60:   required from here
/home/andrew/wwiv/bbs/../bbs/input.h:75:65: error: call of overloaded ‘log10(unsigned char&)’ is ambiguous
   int len = std::max<int>(1, static_cast<T>(ceil(log10(max_value))));
                                                                 ^
/home/andrew/wwiv/bbs/../bbs/input.h:75:65: note: candidates are:
In file included from /usr/include/math.h:32:0,
                 from /usr/gcc/4.9/include/c++/4.9.4/cmath:44,
                 from /usr/gcc/4.9/include/c++/4.9.4/random:38,
                 from /usr/gcc/4.9/include/c++/4.9.4/bits/stl_algo.h:66,
                 from /usr/gcc/4.9/include/c++/4.9.4/algorithm:62,
                 from /home/andrew/wwiv/bbs/conf.cpp:21:
/usr/include/iso/math_iso.h:209:21: note: long double std::log10(long double)
  inline long double log10(long double __X) { return __log10l(__X); }
                     ^
/usr/include/iso/math_iso.h:179:15: note: float std::log10(float)
  inline float log10(float __X) { return __log10f(__X); }
               ^
/usr/include/iso/math_iso.h:81:15: note: double std::log10(double)
 extern double log10 __P((double));
               ^
In file included from /home/andrew/wwiv/bbs/conf.cpp:27:0:
/home/andrew/wwiv/bbs/../bbs/input.h: In instantiation of ‘typename std::enable_if<std::is_unsigned<_Tp>::value, T>::type input_number(T, T, T, bool) [with T = short unsigned int; typename std::enable_if<std::is_unsigned<_Tp>::value, T>::type = short unsigned int]’:
/home/andrew/wwiv/bbs/conf.cpp:623:64:   required from here
/home/andrew/wwiv/bbs/../bbs/input.h:75:65: error: call of overloaded ‘log10(short unsigned int&)’ is ambiguous
   int len = std::max<int>(1, static_cast<T>(ceil(log10(max_value))));
                                                                 ^
/home/andrew/wwiv/bbs/../bbs/input.h:75:65: note: candidates are:
In file included from /usr/include/math.h:32:0,
                 from /usr/gcc/4.9/include/c++/4.9.4/cmath:44,
                 from /usr/gcc/4.9/include/c++/4.9.4/random:38,
                 from /usr/gcc/4.9/include/c++/4.9.4/bits/stl_algo.h:66,
                 from /usr/gcc/4.9/include/c++/4.9.4/algorithm:62,
                 from /home/andrew/wwiv/bbs/conf.cpp:21:
/usr/include/iso/math_iso.h:209:21: note: long double std::log10(long double)
  inline long double log10(long double __X) { return __log10l(__X); }
                     ^
/usr/include/iso/math_iso.h:179:15: note: float std::log10(float)
  inline float log10(float __X) { return __log10f(__X); }
               ^
/usr/include/iso/math_iso.h:81:15: note: double std::log10(double)
 extern double log10 __P((double));
               ^
make[2]: *** [bbs/CMakeFiles/bbs_lib.dir/build.make:567: bbs/CMakeFiles/bbs_lib.dir/conf.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:431: bbs/CMakeFiles/bbs_lib.dir/all] Error 2
make: *** [Makefile:128: all] Error 2
```